[BUG - Develop] Funcparser `$search()` doesn't check access appropriately

#### Describe the bug
This line in funcparser:

https://github.com/evennia/evennia/blob/develop/evennia/utils/funcparser.py#L1165

This seems to be intended to check if `caller` can access `target`. But it seems to always pass, possibly because `access` is meant to be called with two arguments and this invocation is slurping the `"control"` access level as an unused args?

#### To Reproduce
Failing test:

```
    def test_search_no_access(self):
        string = "Go to $search(Room)"
        with self.assertRaises(funcparser.ParsingError):
            self.parser.parse(string, caller=self.char2, return_list=True, raise_errors=True)
```

Failure:
```
======================================================================
FAIL: test_search_no_access (evennia.utils.tests.test_funcparser.TestCallableSearch)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/tmoney/code/evennia/evennia/utils/tests/test_funcparser.py", line 708, in test_search_no_access
    self.parser.parse(string, caller=self.char2, return_list=True, raise_errors=True)
AssertionError: ParsingError not raised

----------------------------------------------------------------------
```

#### Expected behavior
Test passes because `self.char2` doesn't have a "control" access on any rooms.

#### Develop-branch commit
b0f24f9975e14fddeabef06a19b5b61575957b1c

#### Additional context
Again, a small bug that should be easy to fix, but I'd rather submit a skipped failing test and fix it after review.

--- a/evennia/utils/funcparser.py
+++ b/evennia/utils/funcparser.py
@@ -1162,7 +1162,7 @@ def funcparser_callable_search(*args, caller=None, access="control", **kwargs):
         )
 
     for target in targets:
-        if not target.access(caller, target, access):
+        if not target.access(caller, access):
             raise ParsingError("$search Cannot add found entity - access failure.")
 
     return list(targets) if return_list else targets[0]

--- a/evennia/utils/tests/test_funcparser.py
+++ b/evennia/utils/tests/test_funcparser.py
@@ -528,6 +528,7 @@ class TestCallableSearch(test_resources.BaseEvenniaTest):
         """
         string = "$search(TestAccount, type=account)"
         expected = self.account
+        self.account.locks.add("control:id(%s)" % self.char1.dbref)
 
         ret = self.parser.parse(string, caller=self.char1, return_str=False, raise_errors=True)
         self.assertEqual(expected, ret)
@@ -539,6 +540,7 @@ class TestCallableSearch(test_resources.BaseEvenniaTest):
         """
         string = "$search(Script, type=script)"
         expected = self.script
+        self.script.locks.add("control:id(%s)" % self.char1.dbref)
 
         ret = self.parser.parse(string, caller=self.char1, return_str=False, raise_errors=True)
         self.assertEqual(expected, ret)

